src/Nether.Analytics/OutputFormatters/IHeaderAwareOutputFormatter.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Nether.Analytics
+{
+    /// <summary>
+    /// The interface enables the output formatter to be aware of the header. 
+    /// It also provides certain methods that enable features like serializing the 
+    /// message with the header for certain scenarios (i.e. CSV).
+    /// </summary>
+    public interface IHeaderAwareOutputFormatter : IOutputFormatter
```

Comment:
I see where we want to go with this interface and I do like it. The problem I do see is that most OutputManagers never want to use the method FormatWithHeaders, since they only will add a header once per file, blob, etc. Can we perhaps keep the interface, but remove the FormatWithHeaders method and making that part of the specific output managers that need to output every message together with a header. That would keep the implementation very similar and the interface cleaner.